venues/MIDL.amsterdam/2018/Conference/process/recruitReviewerProcess.js

```diff
@@ -1,64 +1,63 @@
-      } else if (note.content.response == 'No'){
-        console.log("Invitation replied No");
-        //if a user is in the reviewers group, remove them from that group and add them to the reviewers-declined group
-        var acceptP = or3client.removeGroupMember('MIDL.amsterdam/2018/Conference/Paper'+paper.number+'/Reviewers/Accepted', note.content.email, token);
-        var declineP = or3client.addGroupMember('MIDL.amsterdam/2018/Conference/Paper'+paper.number+'/Reviewers/Declined', note.content.email, token);
-        var orteamMessage = 'MIDL 2018 Recruitment declination received:\n\n\
+  var orteamMessageDecline = 'MIDL 2018 Recruitment declination received:\n\n\
 navigate to openreview-scripts/venues/MIDL.amsterdam/2018/Conference/python\n\n\
 then run the following:\n\n\
-python assign-reviewers.py ' + note.content.email + ',' + paper.number + ' --remove --baseurl https://openreview.net\n\n\
+python assign-reviewers.py ' + note.content.email + ',' + paperNumber + '\n\n\
 Contact Michael with questions.';
 
-        var orteamMessageBody = {
-          groups: ['info@openreview.net'],
-          subject: '[MIDL 2018] Recruitment Acceptance Received',
-          message: orteamMessage
-        };
-        var messageP = or3client.or3request(or3client.mailUrl, orteamMessageBody, 'POST', token);
-      } else {
-        done('Invalid response', note.content.response);
-      }
 
-      return Promise.all([acceptP, declineP, messageP]);
+  if (hashKey == note.content.key) {
+    if (note.content.response == 'Yes'){
+      var acceptP = or3client.removeGroupMember('MIDL.amsterdam/2018/Conference/' + paperNumber + '/Reviewers/Declined', note.content.email, token);
+      var declineP = or3client.addGroupMember('MIDL.amsterdam/2018/Conference/' + paperNumber + '/Reviewers/Accepted', note.content.email, token);
+
+      var orteamMessageBody = {
+        groups: ['info@openreview.net'],
+        subject: '[MIDL 2018] Recruitment Acceptance Received',
+        message: orteamMessageAccept
+      };
+      var messageP = or3client.or3request(or3client.mailUrl, orteamMessageBody, 'POST', token);
     } else {
-      done('Invalid key', note.content.key);
-      return false;
+      var acceptP = or3client.removeGroupMember('MIDL.amsterdam/2018/Conference/' + paperNumber + '/Reviewers/Accepted', note.content.email, token);
+      var declineP = or3client.addGroupMember('MIDL.amsterdam/2018/Conference/' + paperNumber + '/Reviewers/Declined', note.content.email, token);
+
+      var orteamMessageBody = {
+        groups: ['info@openreview.net'],
+        subject: '[MIDL 2018] Recruitment Acceptance Received',
+        message: orteamMessageDecline
+      };
+      var messageP = or3client.or3request(or3client.mailUrl, orteamMessageBody, 'POST', token);
+
     }
-  })
-  .then(result=>done())
-  .catch(error=>done(error))
 
-  return true
+    Promise.all([acceptP, declineP, messageP])
```

Comment:
promises variables are defined inside the if that is not the scope of the Promise.all. I'm not sure if this is working.

Reply:
my tests were working (thought it's possible I missed something). also, I think this will be ok, because the promise variables are instantiated in both branches of the if statement. either way, the promises will be made